core/lib/Thelia/Controller/Admin/ProductController.php

```diff
@@ -1804,4 +1816,365 @@ protected function arrayHasEntries(array $data, array $entries)
+        foreach ($opAssocConts as $opAssocCont) {
+            $cpProdAssocCont = new ProductAssociatedContent();
+            $cpProdAssocCont
+                ->setProductId($cpId)
+                ->setContentId($opAssocCont->getContentId())
+                ->setPosition($opAssocCont->getPosition())
+                ->save();
+        }
+    }
+
+    public function setCloneI18n($lang, $opId, $cpId)
+    {
+        // Get original product i18n
+        $opI18ns = ProductI18nQuery::create()
+            ->findById($opId);
+
+        /* For each original I18n
+         * - If it's the same language as the one at the product creation, update information
+         * - If not, create a new ProductI18n for the language
+         */
+        foreach ($opI18ns as $opI18n) {
+            if ($opI18n->getLocale() == $lang) {
+                $cpI18n = ProductI18nQuery::create()
+                    ->filterByLocale($lang)
+                    ->findOneById($cpId);
+
+                $cpI18n
+                    ->setChapo($opI18n->getChapo())
+                    ->setDescription($opI18n->getDescription())
+                    ->setPostscriptum($opI18n->getPostscriptum())
+                    ->setMetaTitle($opI18n->getMetaTitle())
+                    ->setMetaDescription($opI18n->getMetadescription())
+                    ->setMetaKeywords($opI18n->getMetaKeywords())
+                    ->save();
+            } else {
+                $cpI18n = new ProductI18n();
+                $cpI18n
+                    ->setId($cpId)
+                    ->setLocale($opI18n->getLocale())
+                    ->setTitle($opI18n->getTitle())
+                    ->setChapo($opI18n->getChapo())
+                    ->setDescription($opI18n->getDescription())
+                    ->setPostscriptum($opI18n->getPostscriptum())
+                    ->setMetaTitle($opI18n->getMetaTitle())
+                    ->setMetaDescription($opI18n->getMetadescription())
+                    ->setMetaKeywords($opI18n->getMetaKeywords())
+                    ->save();
+            }
+        }
+    }
+
+    public function setCloneImage($opId, $cpId)
+    {
+        // Get original product images
+        $opImages = ProductImageQuery::create()
+            ->findByProductId($opId);
+
+        // Set clone's images
+        foreach ($opImages as $opImage) {
+            $cpImage = new ProductImage();
+            $cpImage
+                ->setProductId($cpId)
+                ->setFile($opImage->getFile())
+                ->setVisible($opImage->getVisible())
+                ->setPosition($opImage->getPosition())
+                ->save();
+
+            // Get original product images I18n
+            $opImageI18ns = ProductImageI18nQuery::create()
+                ->findById($opImage->getId());
+
+            // Set clone's images I18n
+            foreach ($opImageI18ns as $opImageI18n) {
+                $cpImageI18n = new ProductImageI18n();
```

Comment:
Have a look at Propel's copy() method, it may saves you from writing tedious code.